controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelper.java

```diff
@@ -106,7 +106,13 @@ static HttpResponse runDetailsResponse(JobController jobController, RunId runId,
         Slime slime = new Slime();
         Cursor logsObject = slime.setObject();
 
-        logsObject.setBool("active", jobController.active(runId).isPresent());
+        try { // Try to fetch new logs, if the run is still active.
+            jobController.updateTestLog(runId);
+            logsObject.setBool("active", true);
+        } // If it isn't, it fails like so.
```

Comment:
This is one of the rare cases where the exception means something else than (give up what you are trying to do), so it may be worth creating a particular exception for this.

IllegalArgumentException, in particular is the one we use to signal that the client (ultimately the user/requestor) is asking for something wrong.

Reply:
You're absolutely right.